File: client-clojure/src/sirqul_io_t_platform/specs/profile_response.clj
(ns sirqul-io-t-platform.specs.profile-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.app-info-response :refer :all]
            [sirqul-io-t-platform.specs.connection-info-response :refer :all]
            [sirqul-io-t-platform.specs.profile-info-response :refer :all]
            )
  (:import (java.io File)))


(def profile-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :loginAccountId) int?
   (ds/opt :validated) boolean?
   (ds/opt :appInfo) app-info-response-spec
   (ds/opt :canViewAppInfo) boolean?
   (ds/opt :canViewFriendInfo) boolean?
   (ds/opt :canViewProfileInfo) boolean?
   (ds/opt :flagCount) int?
   (ds/opt :friendInfo) connection-info-response-spec
   (ds/opt :isBlocked) boolean?
   (ds/opt :isProfileOwner) boolean?
   (ds/opt :metFlagThreshold) boolean?
   (ds/opt :profileInfo) profile-info-response-spec
   (ds/opt :trusted) boolean?
   (ds/opt :following) boolean?
   (ds/opt :friendRequested) boolean?
   (ds/opt :friendRequestPending) boolean?
   (ds/opt :blocked) boolean?
   (ds/opt :flagged) boolean?
   (ds/opt :profileOwner) boolean?
   (ds/opt :friend) boolean?
   (ds/opt :returning) string?
   })

(def profile-response-spec
  (ds/spec
    {:name ::profile-response
     :spec profile-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/ProfileResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ProfileResponse._

case class ProfileResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
loginAccountId: Option[Long],
validated: Option[Boolean],
appInfo: Option[AppInfoResponse],
canViewAppInfo: Option[Boolean],
canViewFriendInfo: Option[Boolean],
canViewProfileInfo: Option[Boolean],
flagCount: Option[Long],
friendInfo: Option[ConnectionInfoResponse],
isBlocked: Option[Boolean],
isProfileOwner: Option[Boolean],
metFlagThreshold: Option[Boolean],
profileInfo: Option[ProfileInfoResponse],
trusted: Option[Boolean],
following: Option[Boolean],
friendRequested: Option[Boolean],
friendRequestPending: Option[Boolean],
blocked: Option[Boolean],
flagged: Option[Boolean],
profileOwner: Option[Boolean],
friend: Option[Boolean],
returning: Option[String])

object ProfileResponse {
  import DateTimeCodecs._

  implicit val ProfileResponseCodecJson: CodecJson[ProfileResponse] = CodecJson.derive[ProfileResponse]
  implicit val ProfileResponseDecoder: EntityDecoder[ProfileResponse] = jsonOf[ProfileResponse]
  implicit val ProfileResponseEncoder: EntityEncoder[ProfileResponse] = jsonEncoderOf[ProfileResponse]
}
